src/Mollie/WC/Helper/PaymentMethodsIconUrl.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+use Mollie\Api\Types\PaymentMethod;
+
+class Mollie_WC_Helper_PaymentMethodsIconUrl
+{
+    /**
+     * @var array
+     */
+    private $paymentMethodImages;
+
+    /**
+     * PaymentMethodIconUrl constructor.
+     * @param array $paymentMethodImages
+     */
+    public function __construct(array $paymentMethodImages)
+    {
+        $this->paymentMethodImages = $paymentMethodImages;
```

Comment:
There's no validation against the values you get in and you assume later that `svg` is a property which exists in the object within the array given to the constructor. 

I would validate the values you get in a way similar to 

```
// Need unit tests with a data provider which test against a set of possible values.
function isArrayAssociative(array $array)
{
    return ([] !== $array) && array_keys($array) !== range(0, count($array) - 1);
}
```

```
public function __construct(array $paymentMethodImages)
{
    $this->bailIfPaymentMethodImagesIsNotValidArgument($paymetMethodImages);
    $this->paymentMethodImages = $paymentMethodImages;
}

// Need test
protected function bailIfPaymentMethodImagesIsNotValidArgument(array $paymentMethodImages)
{
    if(!isArrayAssociative($paymentMethodImages)) {
        $type = gettype($paymentMethodImages);
        throw new InvalidArgumentException(__CLASS__ . " expected an associative array, value of {$type} type given.";
    }

    $validPaymentMethodImages = array_filter($paymentMethodImages, function($paymentMethodImage) {
         return property_exists($paymentMethodImage, 'svg');
     });

     if ($validPaymentMethodImages !== $paymentMethodImages) {
          // May be the message here could be more concise.
          throw new InvalidArgumentException( 
               __CLASS__ . ' expect an associative array of object containing svg properties. Some of the elements within the array does not have the `svg` property.'
          );
     }
}
```

Reply:
After a call I get explained that checking for an associative array will prevent the `fallToAssets` to be called at all.

Best way is to check if the property `svg` isset for the specific request image and fallback if not.